src/simplnx/Utilities/Parsing/DREAM3D/Dream3dIO.cpp

```diff
@@ -883,6 +893,44 @@ Result<IDataArray*> readLegacyDataArray(DataStructure& dataStructure, const nx::
   }
 
   Result<IDataArray*> daResult;
+  // Let's do some sanity checking of the tuple dims and the comp dims
+  // This is here because some 3rd party apps are trying to write .dream3d
+  // files but are getting confused about the Tuple Dims and the component
+  // dims. This bit of code will try to fix the specific issue where the
+  // .dream3d file essentially had _all_ the dimensions shoved in the
+  // tuple dimensions.
+  if(!amTDims.empty()) // If the amTDims are empty we are just reading a Data Array without an Attribute Matrix so skip.
+  {
+    //
+    if(tDims.size() > amTDims.size() && cDims.size() == 1 && cDims[0] == 1)
+    {
+      daResult.warnings().push_back(
+          {-9543, fmt::format(".dream3d file Tuple and Component Dimensions are not correct. DataSet {} does not have correct component dimensions of they are missing.", dataArrayReader.getName())});
+      cDims.clear();
+      // Find the index of the first non-matching dimension value.
+      for(size_t idx = 0; idx < tDims.size(); idx++)
+      {
+        // If we are past the end of the AM Dimensions, then copy the value to the component dimensions
+        if(idx >= amTDims.size())
+        {
+          cDims.push_back(tDims[idx]);
+        }
+        else if(amTDims[idx] != tDims[idx]) // something went wrong here. Each vector should have the same exact values at the start of the vector
+        {
+          // This should not happen at all. If so, bail out now.
+          cDims.clear(); // just put things back the way they were and fail.
+          cDims.push_back(1ULL);
+          daResult.errors().push_back({-9545, fmt::format(".dream3d file Tuple and Component Dimensions are not correct. DataSet {} does not have correct component dimensions of they are missing. An "
```

Comment:
Should we just return here?

Reply:
Code added to return when the error hits.